src/cmds/qstat.c

```diff
@@ -1368,20 +1382,36 @@ display_statjob(struct batch_status *status, struct batch_status *prtheader, int
 				if (a->name != NULL) {
 					if (strcmp(a->name, ATTR_name) == 0) {
 						l = strlen(a->value);
-						if (l >= NAMEL) {
-							strncpy(long_name, a->value, (NAMEL - 1));
-							c = long_name;
-						} else
-							c = a->value;
+						if (wide) {
+							if (l >= SIZEJOBNAME) {
+								strncpy(long_name, a->value, (SIZEJOBNAME - 1));
```

Comment:
Is long_name zeroed?  If it isn't, then only writing N-1 bytes doesn't help you.  It will only write N-1 bytes into the buffer and leave the Nth garbage character alone.

Reply:
It is zeroed:
https://github.com/vstumpf/pbspro/blob/5e7bd1b49888dc3ae825ddd3f702291b5d80de74/src/cmds/qstat.c#L1214